baseband/vdif/tests/test_vdif.py

```diff
@@ -1541,3 +1541,19 @@ def test_find_header_lost_start(self, filename, tmpdir):
             header = fc.find_header()
             assert header is not None
             assert header == h1
+
+def test_edf3_vdif_payload_size():
+    """Test payload size in VDIF EDF3."""
+    testfile = str(tmpdir.join('test.vdif'))
+    with vdif.open(SAMPLE_FILE, 'rb') as fh, vdif.open(testfile, 'wb') as fw:
+        fr = fh.read_frameset()
+        header1 = fr.frames[0].header.copy()
+        header1.payload_nbytes = 1000
+        fr = fr.fromdata(fr.data.reshape(4000, 40, 1), header1)
+        fr.tofile(fw)
+
+    with open(testfile, 'rb') as fc:
+        header2 = fc.read_header()
+        assert header2.payload_nbytes == 1000
+        assert header2.samples_per_frame == 4000
+    
```

Comment:
Here, again open with 'rs', so you can just test `fc.header0`.
Also, might as well check the data is written correctly:
```
data2 = fc.read()
assert np.all(data2 == data1)
```